test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -43,10 +43,42 @@ var (
 
 // StackManager handles
 type StackManager struct {
+	stacks *safeStackMap
+}
+
+type safeStackMap struct {
 	stacks map[string]*auto.Stack
 	lock   sync.RWMutex
 }
 
+func newSafeStackMap() *safeStackMap {
+	return &safeStackMap{stacks: map[string]*auto.Stack{}, lock: sync.RWMutex{}}
+}
+
+func (s *safeStackMap) Get(key string) (*auto.Stack, bool) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	stack, ok := s.stacks[key]
+	return stack, ok
+}
+
+func (s *safeStackMap) Set(key string, value *auto.Stack) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.stacks[key] = value
+}
+
+func (s *safeStackMap) Range(f func(string, *auto.Stack)) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
As you use items inside the map without changing their reference, a read lock should be enough
```suggestion
	s.lock.RLock()
	defer s.lock.RUnlock()
```

Reply:
done